### ISSUE TITLE
xbps-src bootstrap fails when building chroot-git

### ISSUE THREAD

**Issue by suiginsoft** [voidlinux/void-packages#4449]
I'm trying to do my own bootstrap from sources and it fails when it gets to chroot-git, it's trying to link in strlcpy but it doesn't exist in glibc. Something must have broken somewhere.

```
$ ./xbps-src bootstrap
=> base-chroot-0.64_1: building ...
   [runtime] base-files>=0.127: found base-files-0.139_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] kernel-libc-headers>=0: found kernel-libc-headers-4.1.15_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] glibc-devel>=0: found glibc-devel-2.23_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] glibc-locales>=0: found glibc-locales-2.23_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] binutils>=0: found binutils-2.26_3 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] gcc>=0: found gcc-4.9.3_4 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] gcc-c++>=0: found gcc-c++-4.9.3_4 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] patch>=0: found patch-2.7.5_3 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-bash>=0: found chroot-bash-4.3.042_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-grep>=0: found chroot-grep-2.21_2 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] coreutils>=0: found coreutils-8.25_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] gettext>=0: found gettext-0.19.8.1_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-texinfo>=0: found chroot-texinfo-4.13a_8 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] sed>=0: found sed-4.2.2_11 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-gawk>=0: found chroot-gawk-4.0.0_5 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] findutils>=0: found findutils-4.6.0_2 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] diffutils>=0: found diffutils-3.3_8 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] make>=0: found make-4.2.1_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] gzip>=0: found gzip-1.8_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] file>=0: found file-5.28_1 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] tar>=0: found tar-1.28_7 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-util-linux>=0: found chroot-util-linux-2.24.1_4 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-distcc>=0: found chroot-distcc-3.2rc1_4 (/devel/void/void-packages/hostdir/binpkgs)
   [runtime] chroot-git>=0: not found.
   [runtime] ccache>=0: not found.
   [runtime] xbps>=0: not found.
   [runtime] xz>=0: not found.
   [runtime] which>=0: not found.
=> chroot-git-2.7.2_1: building (dependency of base-chroot) ...
   [target] zlib-devel-1.2.8_6: installed.
=> chroot-git: building (dependency of base-chroot) ...
=> chroot-git-2.7.2_1: running pre-build hook: 02-script-wrapper ...
=> chroot-git-2.7.2_1: running do_build ...
    LINK git-credential-store
    LINK git-daemon
    LINK git-fast-import
    LINK git-http-backend
daemon.o: In function `main':
daemon.c:(.text.startup+0xf74): undefined reference to `strlcpy'
libgit.a(path.o): In function `mksnpath':
path.c:(.text+0xa90): undefined reference to `strlcpy'
libgit.a(sha1_name.o): In function `get_sha1_with_context_1':
sha1_name.c:(.text+0x1ed5): undefined reference to `strlcpy'
sha1_name.c:(.text+0x2213): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp':
wrapper.c:(.text+0xa8f): undefined reference to `strlcpy'
libgit.a(wrapper.o):wrapper.c:(.text+0xea2): more undefined references to `strlcpy' follow
collect2: error: ld returned 1 exit status
make: *** [Makefile:1990: git-daemon] Error 1
make: *** Waiting for unfinished jobs....
libgit.a(path.o): In function `mksnpath':
path.c:(.text+0xa90): undefined reference to `strlcpy'
libgit.a(sha1_name.o): In function `get_sha1_with_context_1':
sha1_name.c:(.text+0x1ed5): undefined reference to `strlcpy'
sha1_name.c:(.text+0x2213): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp':
wrapper.c:(.text+0xa8f): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp_mode':
wrapper.c:(.text+0xea2): undefined reference to `strlcpy'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1990: git-http-backend] Error 1
libgit.a(path.o): In function `mksnpath':
path.c:(.text+0xa90): undefined reference to `strlcpy'
libgit.a(sha1_name.o): In function `get_sha1_with_context_1':
sha1_name.c:(.text+0x1ed5): undefined reference to `strlcpy'
sha1_name.c:(.text+0x2213): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp':
wrapper.c:(.text+0xa8f): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp_mode':
wrapper.c:(.text+0xea2): undefined reference to `strlcpy'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1990: git-fast-import] Error 1
libgit.a(path.o): In function `mksnpath':
path.c:(.text+0xa90): undefined reference to `strlcpy'
libgit.a(sha1_name.o): In function `get_sha1_with_context_1':
sha1_name.c:(.text+0x1ed5): undefined reference to `strlcpy'
sha1_name.c:(.text+0x2213): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp':
wrapper.c:(.text+0xa8f): undefined reference to `strlcpy'
libgit.a(wrapper.o): In function `xmkstemp_mode':
wrapper.c:(.text+0xea2): undefined reference to `strlcpy'
collect2: error: ld returned 1 exit status
make: *** [Makefile:1990: git-credential-store] Error 1
=> ERROR: chroot-git-2.7.2_1: failed to run do_build() at line 13.
```

**Comment by xtraeme** [comment]
Fixed with #4514

**Comment by suiginsoft** [comment]
Thanks!
